nanoFramework.DependencyInjection/DependencyInjection/ServiceProviderEngine.cs

```diff
@@ -121,6 +121,51 @@ internal object[] GetService(Type[] serviceType, IServiceCollection scopeService
             return array.Length != 0 ? array : new object[0];
         }
 
+        /// <summary>
+        /// Gets the <see cref="ServiceDescriptor"/> that have been registered for the specified type.
+        /// </summary>
+        /// <param name="serviceType">An object that specifies the type of service object to get.</param>
+        /// <param name="scopeServices">Services collection from current scope.</param>
+        /// <returns>A list of <see cref="ServiceDescriptor"/> that have been registered for the specified type.</returns>
+        /// <exception cref="InvalidOperationException">If <see cref="ServiceProviderOptions.ValidateScopes"/> is true and <see cref="scopeServices"/> is null.</exception>
+        private ServiceDescriptor[] GetServiceDescriptors(Type serviceType, IServiceCollection scopeServices = null)
+        {
+            var descriptors = new ArrayList();
+
+            if (scopeServices is not null)
+            {
+                foreach (ServiceDescriptor descriptor in scopeServices)
+                {
+                    if (descriptor.ServiceType != serviceType) continue;
+
+                    descriptors.Add(descriptor);
+                }
+            }
+
+            foreach (ServiceDescriptor descriptor in Services)
+            {
+                if (descriptor.ServiceType != serviceType) continue;
+
+                switch (descriptor.Lifetime)
+                {
+                    case ServiceLifetime.Singleton:
+                        descriptors.Add(descriptor);
+                        break;
+
+                    case ServiceLifetime.Transient:
```

Comment:
Can we "merge" both cases to stress they should be handled the same way?

Reply:
Yes we can 👍